megamek/src/megamek/client/commands/FireCommand.java

```diff
@@ -294,6 +295,13 @@ private String calculateToHit(int weaponId, Targetable target) {
             			|| (m.getType().hasModes() && m.curMode().equals("Point Defense"))) {
                 str += " Can't shoot: "
                        + Messages.getString("FiringDisplay.autoFiringWeapon");
+            } else if (getClient().getGame().getPhase() == IGame.Phase.PHASE_FIRING 
+                        && ((m.getType().hasModes() && m.curMode().equals("Bearings-Only Extreme Detection Range"))
```

Comment:
I notice a lot of these string comparison checks for extreme/long/medium/short detection range. Is there a way you could consolidate them into a single function? Looking at the curMode(), it doesn't look like you can escape having to do a bunch of string comparisons (which is unfortunate, but let's leave it alone for now), but I we can save quite a bit of code volume at least.